src/test/java/de/zalando/aruha/nakadi/repository/kafka/KafkaTopicRepositoryTest.java

```diff
@@ -256,6 +257,17 @@ public void canGetPartition() throws NakadiException {
     }
 
     @Test
+    public void testIntegerOverflowOnStatisticsCalculation() throws NakadiException {
+        when(settings.getMaxTopicPartitionCount()).thenReturn(1000);
+        final EventTypeStatistics statistics = new EventTypeStatistics();
+        statistics.setReadParallelism(1);
+        statistics.setWriteParallelism(1);
+        statistics.setMessagesPerMinute(1000000000);
+        statistics.setMessageSize(1000000000);
+        assertThat(kafkaTopicRepository.calculateKafkaPartitionCount(statistics), equalTo(6));
```

Comment:
According to collected data - It's not so. The speed is not increasing, so it will select the fastest partition count